activerecord/lib/active_record/inheritance.rb

```diff
@@ -162,14 +162,40 @@ def abstract_class?
         defined?(@abstract_class) && @abstract_class == true
       end
 
+      # Returns the value to be stored in the inheritance column for STI.
       def sti_name
         store_full_sti_class ? name : name.demodulize
       end
 
+      # Returns the class for the provided +type_name+.
+      #
+      # It is used to find the class correspondent to the value stored in the inheritance column.
+      def sti_class_for(type_name)
+        if store_full_sti_class
+          ActiveSupport::Dependencies.constantize(type_name)
```

Comment:
While we're here, is this actually Zeitwerk compatible? cc @fxn

Reply:
Yes, we decorate it https://github.com/rails/rails/blob/02723536f88be1a389908e53239a65680a6147b8/activesupport/lib/active_support/dependencies/zeitwerk_integration.rb#L18

Reply:
:+1: